### ISSUE TITLE
Effects on all leds instead of only the affected ones

### ISSUE THREAD

**Issue by wolph** [SmartHome-yourself/sonoff-tx-ultimate-for-esphome#11]
Hi,

It would be much nicer (especially with the rainbow effect) if the effect would be visible on all of the LEDs similar to how the stock firmware works.

Perhaps you've seen it, but quite a while ago I've made an esphome component and script that does this: https://gist.github.com/wolph/42024a983e4dfb0bc1dcbe6882979d21#file-relays_light-yaml

But your script looks like a more well rounded project than mine so I wouldn't mind switching :)

**Comment by danielscheidler** [comment]
Atm. I'm running out of time, but I'll play around with more animations as soon as I get some rest. 
Maybe we can work together on this project?

**Comment by wolph** [comment]
I've done many tests using both this code and my own version of the code, but no matter what I do, as soon as I start handling all led's the device becomes unresponsive once in a while.
After that I need to remove the power for a while before it starts working again.

I'm not entirely sure what the issue is, but it's related to esphome since I don't have the issue with tasmota. It seems that (for me at least) both esphome and tasmota have a few (dis)advantages so I've been playing around with both, but esphome doesn't seem workable for me due to the stability issues I'm getting with this hardware. I've been running esphome on many devices for a few years now without any issues, but for some reason I can't get this thing (I've got 3 tx ultimates, all the same issue) to work reliably.

Advantages of esphome:
 - prettier/nicer led effects
 - easier to configure with a simple yaml file (once someone else has created the package)
 - easier to upload/update with firmware updates

Advantages of tasmota:
 - stable, esphome stops responding after a few tests with the leds (I'm guessing the pretty esphome effects cause a brownout)
 - filesystem support so you can easily include sounds
 - more standalone support (i.e. have multiple tasmota devices connect without having to go through mqtt/home-assistant)

I've spent quite a bit of time trying to get esphome to run reliably on these things, but so far I've come up empty... it just keeps giving me issues. I'm just going to keep running tasmota for now unless someone has a fix to make these things respond reliably using esphome.

And esphome doesn't have an easy way to include sounds which is also a big downside for me. With my tasmota version I currently have a mario powerup and mario pipe sound effect for turning the lights on/off respectively ;)

**Comment by danielscheidler** [comment]
So your switch gets stop responding if you use leds on the main-light including all leds? I can imagine, that it could be a problem, that the partitions/segments overlaps and working againgst each other if you use the partitions, segments and main light all at once. 
Atm. I finalize the cover, after that I can try to take a look at using all LEDs for effects too. 
Ok, local audio wasn't a thing for me. For notifications I'm fine with rtttl and for other stuff I am happy with a notify-automation in my Home Assistant. 
But nice to know that this is possible with tasmota and that you point out the differences.

**Comment by wolph** [comment]
Yes, that's been an issue with running esphome on these devices since the beginning for me.

I've had these devices since april and I've done quite a bit of testing but for some reason with esphome I've seen this issue when trying to address all leds. I don't think the partitions/segments are part of the issue but I am not entirely certain... but it's been quite a frustrating thing to debug.

In any case, I think you have done amazing work here! It looks really nice out of the box :)
